lib/jsanalyze.js

```diff
@@ -181,16 +186,52 @@ exports.analyzeJs = function analyzeJs(contents, opts) {
 			comments: false
 		});
 
-		options.presets.push([ babili, {
+		options.presets.push([ minify, {
 			mangle: false,
 			deadcode: false
 		} ]);
 
-		options.plugins.push(require.resolve('babel-plugin-transform-property-literals'));
+		options.plugins.push(require.resolve('@babel/plugin-transform-property-literals'));
 	}
 
 	if (options.presets.length) {
-		// FIXME we can't re-use the ast here, because we traversed it
+
+		// rollup is required to workaround circular references unsupported by JavascriptCore
+		// this should only run on 'app.js', bundling all dependencies into one file
+		if (opts.rollup) {
+			let input = {
+				[opts.filename]: contents
+			};
+
+			// include regenerator-runtime needed for async support
+			if (opts.rollup.runtime) {
+				const runtimePath = path.join(path.dirname(regenerator), 'runtime.js');
+				input[opts.filename] = `import 'runtime';${contents}`;
+				input.runtime = fs.readFileSync(runtimePath).toString();
+			}
+
+			// configure rollup
+			const bundle = await rollup.rollup({
+				input: input,
+				context: `'${path.join(opts.projectDir, 'Resources')}'`,
+				plugins: [
+					resolver()
+				]
+			});
+
+			// generate code and a source map
+			// TODO: make use of source map
+			const { code, map } = await bundle.generate({
```

Comment:
My concern here is that babel has generated bad source maps in the past for us, so I've been trying to work around the issue by using the `retainLines: true` option whenever we pass through babel.

Studio uses the source maps for debugging, but we've found out the hard way that babel's source map generation is not reliable. (We consume alloy's source maps)

It looks as though rollup actually uses acorn, not babel - so maybe it's more reliable?

We need to actually look for and combine the possible alloy source map into the process too.